union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
+	}
+	if len(fileSystems) == 1 {
+		return fileSystems[0]
+	}
+
+	ns := vfs.NameSpace{}
+	for _, fs := range fileSystems {
+		ns.Bind("/", fs, "/", vfs.BindAfter)
+	}
+	return &unionFs{
+		NameSpace:   ns,
+		fileSystems: fileSystems,
+	}
+}
+
+type unionFs struct {
+	vfs.NameSpace
+	fileSystems []FileSystem
+}
+
+func (fs *unionFs) resolve(path string) FileSystem {
+	parent := pathpkg.Dir(path)
+	for _, fs2 := range fs.fileSystems {
+		if _, err := fs2.Stat(parent); err == nil {
+			return fs2
+		}
+	}
+	return fs.fileSystems[0]
+}
+
+// Create creates the named file, truncating it if it already exists.
+func (fs *unionFs) Create(path string) (io.WriteCloser, error) {
+	return fs.resolve(path).Create(path)
```

Comment:
I saw that race condition, but you can't fix it with a mutex here, since the underlying file systems may also be changed from somewhere else, even other processes in case of OS file systems. Synchronization (if possible) needs to be done at a higher level, e.g. the consumer of this API. Maybe the documentation should state that explicitly?